private/bufpkg/bufmodule/module.go

```diff
@@ -56,11 +61,45 @@ func newModuleForProto(
 	if err != nil {
 		return nil, err
 	}
+	protoBreakingConfig := protoModule.GetBreakingConfig()
+	var breakingConfig *bufbreakingconfig.Config
+	if protoBreakingConfig.GetVersion() == bufconfig.V1Version {
+		var err error
+		breakingConfig, err = bufbreakingconfig.NewConfigV1ForProto(protoBreakingConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if protoBreakingConfig.GetVersion() == bufconfig.V1Beta1Version {
+		var err error
+		breakingConfig, err = bufbreakingconfig.NewConfigV1Beta1ForProto(protoBreakingConfig)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
It's impossible for both of these `if` conditions to be true, so we can better convey this with either a `switch protoBreakingConfig.GetVersion()`, or by encapsulating all of this in a simple helper function that returns the `*bufbreakingconfig.Config` type like so:

```go
func breakingConfigForProto(protoBreakingConfig *breakingv1.Config) (*bufbreakingconfig.Config, error) {
  if protoBreakingConfig == nil {
    // Not all modules will have a *breakingv1.Config, so we
    // don't validate against it if it doesn't exist.
    return nil, nil
  }
  switch version := protoBreakingConfig.GetVersion(); version {
    case bufconfig.V1Beta1Version:
      return bufbreakingconfig.NewConfigV1Beta1ForProto(protoBreakingConfig)
    case bufconfig.V1Version:
      return bufbreakingconfig.NewConfigV1ForProto(protoBreakingConfig)
    default:
      return nil, fmt.Errorf("invalid version %q found in proto breaking configuration", version)
  }
}
```

That way you can simplify the error handling here to the following:
```go
breakingConfig, err := breakingConfigForProto(protoModule.GetBreakingConfig())
if err != nil {
  return nil, err
}
lintConfig, err := lintConfigForProto(protoModule.GetLintConfig())
if err != nil {
  return nil, err
}
return newModule(
  ...
)
```

You could also argue that the `breakingConfigForProto` function could be in the `bufbreakingconfig` package, which would let you un-export `bufbreakingconfig.NewConfig{V1Beta1,V1}ForProto`.

Reply:
+1 on the switch. As for rolling them up into a single `NewConfigForProto` function, I think I prefer it to follow the current model for the external config (the separate functions for v1 and beta1) so that it's consistent (even though it's slightly repetitive code, the pattern already exists).